src/components/course-settings/rename-course.cjsx

```diff
@@ -0,0 +1,94 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+{CourseStore, CourseActions} = require '../../flux/course'
+{TutorInput} = require '../tutor-input'
+
+RenameCourseField = React.createClass
+
+  displayName: 'RenameCourseField'
+  propTypes:
+    courseId: React.PropTypes.string
+    label: React.PropTypes.string.isRequired
+    name: React.PropTypes.string.isRequired
+    default: React.PropTypes.string.isRequired
+    onChange: React.PropTypes.func.isRequired
+    autofocus: React.PropTypes.bool
+    validate: React.PropTypes.func.isRequired
+
+  componentDidMount: ->
+    @refs.input.focus() if @props.autofocus
+    @refs.input.cursorToEnd() if @props.autofocus
+
+  onChange: (value) ->
+    @props.onChange(value)
+
+  render: ->
+    <TutorInput
+      ref="input"
+      label={@props.label}
+      default={@props.default}
+      required={true}
+      onChange={@onChange}
+      validate={@props.validate}
+      />
+
+module.exports = React.createClass
```

Comment:
JSX automatically gives the react component the displayName of the left-hand-side of the assignment. so you can just do `RenameCourse = React.createClass` and then later `module.exports = RenameCourse`